### PR TITLE
fix: improve the rendering of build errors when bundling

### PR DESCRIPTION
**Pull request by petebacondarwin**
## What this PR solves / how to test

Fixes #0000

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [x] Tests included
  - [ ] Tests not necessary because:
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [ ] Required
  - [x] Not required because: doesn't affect e2e use cases
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [x] Changeset included
  - [ ] Changeset not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [x] Documentation not necessary because: polish / bug fix

### CODE REVIEW

File: packages/wrangler/src/deployment-bundle/bundle.ts [hunk 1]
```diff
@@ -467,7 +467,11 @@ export async function bundleWorker(
 			await ctx.watch();
 			result = await initialBuildResultPromise;
 			if (result.errors.length > 0) {
-				throw new UserError("Failed to build");
```
Comment (petebacondarwin) [inline packages/wrangler/src/deployment-bundle/bundle.ts:470]: We were just swallowing the errors and warnings here and relying upon them being logged "directly" to stdout by another instance of esbuild that is doing the format sniffing. See below.

File: packages/wrangler/src/deployment-bundle/guess-worker-format.ts [hunk 2]
```diff
@@ -38,6 +38,7 @@ export default async function guessWorkerFormat(
 		metafile: true,
 		bundle: false,
 		write: false,
+		logLevel: "silent",
```
Comment (petebacondarwin) [inline packages/wrangler/src/deployment-bundle/guess-worker-format.ts:41]: When guessing the worker format, we use esbuild but were forgetting to silence its stdout logging.
This led to duplicate logs and also confused things where we were relying on this run of esbuild to output initial build failures.

**Comment by irvinebroque** [comment]
🙌

File: packages/wrangler/src/deployment-bundle/guess-worker-format.ts (hunk 2 above)
Comment (petebacondarwin) [reply packages/wrangler/src/deployment-bundle/guess-worker-format.ts]: Landed in a previous PR.

**Review by emily-shen — APPROVED** [review]

**Pull request merged**